frontend/src/components/uam/components/users-table/users-table.tsx

```diff
@@ -7,22 +7,25 @@ import {
   useAppDispatch,
   useAppSelector,
   useEffect,
-  useState,
+  usePagination,
+  useSearchParams,
 } from 'hooks/hooks';
 import { Column } from 'react-table';
 import { uamActions } from 'store/actions';
 
 import styles from './styles.module.scss';
 
 const UsersTable: FC = () => {
-  const [page, setPage] = useState<number>(PaginationDefaultValue.DEFAULT_PAGE);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const pageFromQuery = Number(searchParams.get('page'));
```

Comment:
```suggestion
```
you need to pass query name to the hook and hook should be able to handle all logic